python/lib/client/dmod/client/request_clients.py

```diff
@@ -1,600 +1,823 @@
+            (Unused) variable keyword args.
 
         Returns
         -------
+        BasicResultIndicator
+            An indicator that, when successful, includes as ``data`` the list of ids of existing jobs.
+
+        See Also
+        -------
+        get_jobs_list
+        """
+        # TODO: implement
+        raise NotImplementedError('{} function "request_jobs_list" not implemented yet'.format(self.__class__.__name__))
+
+    async def submit_ngen_request(self, **kwargs) -> NGENRequestResponse:
+        return NGENRequestResponse.factory_init_from_deserialized_json(
+            json.loads(await self._submit_job_request(request=NGENRequest(request_body=kwargs, **kwargs))))
 
+    async def submit_ngen_cal_request(self, **kwargs) -> NgenCalibrationResponse:
+        return NgenCalibrationResponse.factory_init_from_deserialized_json(
+            json.loads(await self._submit_job_request(request=NgenCalibrationRequest(request_body=kwargs, **kwargs))))
+
+    async def submit_request_from_file(self, job_type: str, request_file: Path, *kwargs) -> ResultIndicator:
         """
+        Submit a serialized job request stored in the given file.
+
+        Parameters
+        ----------
+        job_type : str
+            String representation of the type of request: either ``ngen`` or ``ngen_cal``.
+        request_file : Path
+            The supplied file containing a JSON string, which should be a serialized, applicable request object.
+        """
+        return await self.submit_request_from_json(job_type=job_type, request_json=request_file.read_text())
+
+    async def submit_request_from_json(self, job_type: str, request_json: Union[dict, str], **kwargs) -> ResultIndicator:
+        """
+        Submit a supplied, serialized job request.
+
+        Parameters
+        ----------
+        job_type : str
+            String representation of the type of request: either ``ngen`` or ``ngen_cal``.
+        request_json : Union[dict, str]
+            The serialized representation of a request as JSON, either as a ``str`` or already inflated to a JSON
+            ``dict`` object.
+        """
+        if isinstance(request_json, str):
+            request_json = json.loads(request_json)
+
+        if job_type == 'ngen':
+            return NGENRequestResponse.factory_init_from_deserialized_json(
+                json.loads(await self._submit_job_request(NGENRequest.factory_init_from_deserialized_json(request_json))))
+        elif job_type == 'ngen_cal':
+            return NgenCalibrationResponse.factory_init_from_deserialized_json(
+                json.loads(await self._submit_job_request(NgenCalibrationRequest.factory_init_from_deserialized_json(request_json))))
+        else:
+            raise RuntimeError(f"Invalid job type indicator for serialized job request: {job_type}")
+
+
+class DataTransferAgent(ABC):
+
+    @abstractmethod
+    async def download_dataset_item(self, dataset_name: str, item_name: str, dest: Path):
```

Comment:
Indeed, I had a similar idea in mind, and even started a bit of work for it, but I decided it would be too much trouble for the time being to finish it correctly.  I have created #432 to track the task in the future.

Reply:
Awesome, thanks for doing that!